File: gilded-rose-ts/test-gilded-rose.ts
import { Item, updateQuality } from "./gilded-rose";
import product from "./product";

product(
    [
        "name",
        "Aged Brie",
        "Backstage passes to a TAFKAL80ETC concert",
        "Sulfuras, Hand of Ragnaros",
    ],
    [1, 0, 11, -1, 6],
    [1, 50, 0, 49]
).forEach(([name, sellIn, quality]) => {
    let item: Item = { name, sellIn, quality };

    const before = `${item.name} ${item.sellIn} ${item.quality}`;

    updateQuality([item]);

    const after = `${item.name} ${item.sellIn} ${item.quality}`;

    console.log(`${before} -> ${after}`);
});

File: gilded-rose-with-typist/test_gilded_rose.py
import sys
import os
sys.path.append(os.path.dirname(os.path.abspath(__file__)))

from gilded_rose import GildedRose, Item  # noqa: E402


def main():
    names = ["Aged Brie", "Sulfuras, Hand of Ragnaros",
             "Backstage passes to a TAFKAL80ETC concert", "Regular Item"]
    sell_in_values = [5, 0, 15, 10]
    qualities = [10]

    items = []
    for name in names:
        for sell_in in sell_in_values:
            for quality in qualities:
                items.append(Item(name, sell_in, quality))

    gilded_rose = GildedRose(items)

    gilded_rose.update_quality()

    for item in items:
        print(item)


if __name__ == "__main__":
    main()
